eZ/Bundle/EzPublishCoreBundle/Tests/EventListener/ViewControllerListenerTest.php

```diff
@@ -151,6 +158,63 @@ public function testGetControllerMatchedView()
         $this->assertEquals($expectedView, $this->request->attributes->get('view'));
     }
 
+    public function testGetControllerEmitsProperEvents(): void
+    {
+        $viewObject = new class() extends BaseView {
+        };
+
+        $this->viewBuilderRegistry
+            ->expects($this->once())
+            ->method('getFromRegistry')
+            ->will($this->returnValue($this->viewBuilderMock));
+
+        $this->viewBuilderMock
+            ->expects($this->once())
+            ->method('buildView')
+            ->will($this->returnValue($viewObject));
+
+        $eventDispatcher = new TraceableEventDispatcher(
+            new EventDispatcher(),
+            new Stopwatch()
+        );
+
+        // FilterViewBuilderParametersEvent
+        $eventDispatcher->addListener(
+            ViewEvents::FILTER_BUILDER_PARAMETERS,
+            static function (FilterViewBuilderParametersEvent $event): void {
+                self::assertTrue(true);
+            }
+        );
+
+        // PostBuildViewEvent
+        $eventDispatcher->addListener(
+            PostBuildViewEvent::class,
+            static function (PostBuildViewEvent $event): void {
+                self::assertTrue(true);
+            }
+        );
```

Comment:
The event dispatcher configuration here seems unnecessary. You could instead replace it with `EventDispatcherInterface` mock configured to expect being called with specific values specific amount of times.

This would make it so you don't need to check the internals of `TraceableEventDispatcher`, don't need the `addListener` calls with empty, always true assertions.

```suggestion
        $eventDispatcher = $this->createMock(EventDispatcherInterface::class);
        $eventDispatcher
            ->expect($this->once())
            ->method('dispatch')
            ->withConsecutive(
                [
                    $this->assertInstanceOf(ViewEvent::class),
                    $this->assertSame(ViewEvents::FILTER_BUILDER_PARAMETERS),
                ], [
                    $this->assertInstanceOf(PostBuildViewEvent::class),
                    $this->assertSame(PostBuildViewEvent::class),
                ])
            ->willReturnArgument(0);
```